crates/primitives/starknet/src/crypto/hash/pedersen.rs

```diff
@@ -32,33 +34,17 @@ impl HasherT for PedersenHasher {
     /// # Returns
     ///
     /// The hash of the data.
-    fn hash_elements(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper {
-        let mut hash = FieldElement::ZERO;
-        for element in data {
-            hash = pedersen_hash(&hash, &element.0);
-        }
-
-        let data_len = Felt252Wrapper::from(data.len() as u64);
-        hash = pedersen_hash(&hash, &data_len.0);
-
+    fn compute_hash_on_wrappers(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper {
+        let hash = Self::compute_hash_on_elements(&data.iter().map(|x| x.0).collect::<Vec<FieldElement>>());
         Felt252Wrapper(hash)
     }
-}
 
-impl DefaultHasher for PedersenHasher {
-    fn hasher() -> Self {
-        Self::default()
-    }
-}
-
-/// The pedersen CryptoHasher implementation.
-impl CryptoHasherT for PedersenHasher {
     #[inline(always)]
-    fn hash(a: FieldElement, b: FieldElement) -> FieldElement {
+    fn hash_elements(a: FieldElement, b: FieldElement) -> FieldElement {
         pedersen_hash(&a, &b)
     }
 
-    /// Compute hash on elements, base on the [python implementation](https://github.com/starkware-libs/cairo-lang/blob/12ca9e91bbdc8a423c63280949c7e34382792067/src/starkware/cairo/common/hash_state.py#L6-L15).
+    /// Compute hash on elements, taken from [starknet-rs](https://github.com/xJonathanLEI/starknet-rs/blob/master/starknet-core/src/crypto.rs#L25) pending a no_std support.
```

Comment:
why not use it directly then ?

Reply:
`starknet_core` only works with `std`, we could have conditional compilation and do

```
#[cfg(feature = "std")]
use starknet_core::crypto::compute_hash_on_elements;
```

But I think it will require to have two diff. impls. (std and no_std) for functions where we want to use `compute_hash_on_elements`; std would use it and then no_std a "custom" impl.

Somewhat new to Rust so not sure ..

Reply:
Hm I think starknet-rs is no_std compatible since we do use it everywhere in practice

Reply:
Not all the crates! There is actually an issue to make `starknet-core` compatible with no_std [here](https://github.com/xJonathanLEI/starknet-rs/issues/403), @dbejarano820 is working on it 🙏